### PR TITLE
Separate test commands

### PR DESCRIPTION
**Pull request by karol-bisztyga**

### CODE REVIEW

File: protostar/commands/cairo1_commands/test_cairo1_command.py [hunk 1]
```diff
@@ -108,32 +118,95 @@ async def run(self, args: Namespace) -> TestingSummary:
...
+    ) -> TestingSummary:
+        include_paths = [
+            str(path)
+            for path in self._project_cairo_path_builder.build_project_cairo_path_list(
+                cairo_path or []
+            )
+        ]
+        include_paths.append(
+            str(self._protostar_directory.protostar_cairo1_corelib_path)
+        )
```
Comment (Arcticae) [inline protostar/commands/cairo1_commands/test_cairo1_command.py:154]: [nitpick]: If it's always appended it doesn't make sense to have it in a seprarate line

**Review by Arcticae — APPROVED** [review]
Can you rebase/change the base of the PR? It contains deploy PR changes.

Also tagging @MaksymilianDemitraszek to make sure it's what he wanted.


Apart from the nitpicks and the bases, it looks good to me.

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py [hunk 2]
```diff
@@ -23,6 +23,11 @@ def get_mock_for_lib_func(
         return_value = type(
             "return_value", (object,), {"err_code": err_code, "ok": ok}
         )()
+    elif lib_func_name == "deploy":
```
Comment (MaksymilianDemitraszek) [inline tests/integration/cairo1_compiler_bindings/library_functions_test.py:26]: This seems like unrelated chagne

**Review by MaksymilianDemitraszek — APPROVED** [review]

File: protostar/commands/cairo1_commands/test_cairo1_command.py [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
+    TestScheduler,
+    determine_testing_seed,
 )
+from protostar.io.output import Messenger
 
-from protostar.commands.test import TestCommandCache
+from ..test.test_command_cache import TestCommandCache
+from ..test.messages import TestCollectorResultMessage
+from ...cairo_testing.cairo1_test_collector import Cairo1TestCollector
+from ...cairo_testing.cairo1_test_runner import Cairo1TestRunner
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply protostar/commands/cairo1_commands/test_cairo1_command.py]: [make imports absolute](https://github.com/software-mansion/protostar/pull/1602/commits/2cb49be44a65d8123cc50fc6c7ae9af09f83ad76)

File: protostar/commands/cairo1_commands/test_cairo1_command.py [hunk 4]
```diff
@@ -108,32 +118,95 @@ async def run(self, args: Namespace) -> TestingSummary:
...
+        include_paths.append(
+            str(self._protostar_directory.protostar_cairo1_corelib_path)
+        )
+
+        testing_seed = determine_testing_seed(seed=None)
+
+        with ActivityIndicator(
+            self._log_color_provider.colorize("GRAY", "Collecting tests")
+        ):
+            compiler_config = CairoCompilerConfig(
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply protostar/commands/cairo1_commands/test_cairo1_command.py]: [7b2ed08](https://github.com/software-mansion/protostar/pull/1602/commits/7b2ed08fb9b1ac654f6b62833c69b962c7104422)

**Comment by MaksymilianDemitraszek** [comment]
@karol-bisztyga Is it possible to merge this 🙏 🥺 we are going to do some changes in related parts of the code and I want to avoid conflicts if possible.

**Comment by karol-bisztyga** [comment]
Yes, I'm only waiting for the CI
